Frameworks/Foundation/NSException.mm

```diff
@@ -266,15 +268,9 @@ - (HRESULT)_hresult {
 
 - (void)_processException {
     HRESULT hr = [self _hresult];
-    assert(hr != S_OK);
-
-    if(self.reason != nil) {
-        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
-        Microsoft::WRL::Wrappers::HStringReference hstrRef(reason);
-        RoOriginateError(hr, hstrRef.Get());
-    } else {
-        RoOriginateError(hr, nullptr);
-    }
+    auto reason = Strings::NarrowToWide<HSTRING>(self.reason);
```

Comment:
nit: I'm not sure that `auto` is good here since the `<HSTRING>` suggests it's an HSTRING rather than an HString